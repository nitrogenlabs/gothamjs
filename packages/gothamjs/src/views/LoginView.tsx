import Button from '@material-ui/core/Button/Button';
import Card from '@material-ui/core/Card/Card';
import CardContent from '@material-ui/core/CardContent/CardContent';
import Grid from '@material-ui/core/Grid/Grid';
import {StyleRulesCallback} from '@material-ui/core/styles';
import {Flux} from '@nlabs/arkhamjs';
import * as React from 'react';

import {Form} from '../components/Form/Form';
import {TextField} from '../components/Form/TextField';
import {PageHeader} from '../components/PageHeader';
import {UserConstants} from '../constants/UserConstants';
import {LoginViewProps, LoginViewState} from '../types/views/login';
import {initComponent} from '../utils/components';
import {PageView} from './PageView';

const styles: StyleRulesCallback = (theme) => ({
  buttonBox: {
    display: 'flex',
    marginBottom: 15,
    marginTop: 30
  },
  buttonRowLeft: {
    display: 'flex'
  },
  buttonRowRight: {
    alignItems: 'flex-end',
    display: 'flex',
    justifyContent: 'flex-end'
  },
  card: {
    borderRadius: 10,
    maxWidth: 750
  },
  forgot: {
    color: theme.palette.grey['400'],
    fontSize: 12,
    marginTop: 5
  },
  logo: {
    alignItems: 'center',
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    marginBottom: 50,
    marginTop: 30
  }
});

export class LoginViewBase extends React.PureComponent<LoginViewProps, LoginViewState> {
  state: any = {};

  constructor(props) {
    super(props);

    // Methods
    this.onFailure = this.onFailure.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onSuccess = this.onSuccess.bind(this);

    // Initial state
    this.state = {
      authentication: {},
      session: Flux.getState('user.session')
    };
  }

  componentWillMount(): void {
    // Add listeners
    Flux.on(UserConstants.LOGIN_SUCCESS, this.onSuccess);
    Flux.on(UserConstants.LOGIN_FAILED, this.onFailure);
  }

  componentWillUnmount(): void {
    // Remove listeners
    Flux.off(UserConstants.LOGIN_SUCCESS, this.onSuccess);
    Flux.off(UserConstants.LOGIN_FAILED, this.onFailure);
  }

  onSuccess(): void {

  }

  onFailure(): void {

  }

  onSubmit(values): void {
    this.setState({authentication: values});
  }

  renderLogo(logo: JSX.Element, classes): JSX.Element {
    if(logo) {
      return <div className={classes.logo}>{logo}</div>;
    }

    return null;
  }

  render(): JSX.Element {
    const {classes, logo, name = 'login', title = 'Login'} = this.props;

    return (
      <PageView name={name} title={title}>
        <Grid item sm={12}>
          <PageHeader title="Welcome Back">
            Welcome back sign-in now there is lot of new stuff waiting for you.
          </PageHeader>
        </Grid>

        <Grid item xs={12}>
          <Grid container alignItems="center" justify="center">
            <Card className={classes.card}>
              <CardContent>
                {this.renderLogo(logo, classes)}
                <Form onSubmit={this.onSubmit}>
                  <Grid item xs={12}>
                    <TextField
                      fullWidth
                      label="Username"
                      name="username"
                      type="text" />
                  </Grid>
                  <Grid item xs={12}>
                    <TextField
                      fullWidth
                      label="Password"
                      name="password"
                      type="password" />
                  </Grid>
                  <Grid className={classes.buttonBox} container>
                    <Grid item xs={12} className={classes.buttonRowLeft}>
                      <p className={classes.forgot}>Forgot password?</p>
                    </Grid>
                    <Grid item xs={12} className={classes.buttonRowRight}>
                      <Button color="default" size="large" type="submit" variant="text">New Account</Button>
                      <Button color="primary" size="large" type="submit" variant="contained">Sign in</Button>
                    </Grid>
                  </Grid>
                </Form>
              </CardContent>
            </Card>
          </Grid>
        </Grid>
      </PageView>
    );
  }
}

export const LoginView = initComponent(module, LoginViewBase, styles);
export default LoginView;
