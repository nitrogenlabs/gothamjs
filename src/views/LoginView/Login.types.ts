/**
 * Copyright (c) 2018-Present, Nitrogen Labs, Inc.
 * Copyrights licensed under the MIT License. See the accompanying LICENSE file for terms.
 */
import {PageViewProps} from '../PageView/PageView.types';

export interface LoginViewProps extends PageViewProps {
  readonly logo?: JSX.Element;
  readonly onLogin?: (username: string, password: string) => any;
  readonly onSignup?: () => any;
}

export interface LoginViewState {
  readonly authentication?: object;
  readonly errorMessage?: string;
  readonly session?: object;
}
