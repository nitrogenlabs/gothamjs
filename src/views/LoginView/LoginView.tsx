/**
 * Copyright (c) 2018-Present, Nitrogen Labs, Inc.
 * Copyrights licensed under the MIT License. See the accompanying LICENSE file for terms.
 */
import Button from '@material-ui/core/Button/Button';
import Card from '@material-ui/core/Card/Card';
import CardContent from '@material-ui/core/CardContent/CardContent';
import {makeStyles} from '@material-ui/styles';
import {useFlux, useState} from '@nlabs/arkhamjs-utils-react/lib';
import * as React from 'react';

import {Form} from '../../components/Form/Form';
import {TextField} from '../../components/Form/TextField';
import {PageHeader} from '../../components/PageHeader/PageHeader';
import {Theme} from '../../config/theme.types';
import {LoginConstants} from '../../constants/LoginConstants';
import {PageView} from '../PageView/PageView';
import {LoginViewProps} from './LoginView.types';

const useStyles: any = makeStyles((theme: Theme) => ({
  btn: {
    marginLeft: 5
  },
  card: {
    borderRadius: 10
  },
  errorMessage: {
    backgroundColor: theme.palette.error.light,
    borderRadius: 3,
    color: theme.palette.error.dark,
    fontSize: '14px',
    marginBottom: 15,
    padding: '10px 5px'
  },
  forgot: {
    color: theme.palette.grey['400'],
    fontSize: 12,
    marginTop: 5
  },
  logo: {
    alignItems: 'center',
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    marginBottom: 50,
    marginTop: 30
  }
}));

export const onSuccess = (): void => {
};

export const onFailure = (setState) => ({error}): void => {
  const {code} = error;
  let errorMessage: string;

  switch(code) {
    case 'UserNotFoundException': {
      errorMessage = 'Username and/or password was not found.';
      break;
    }
    case 'UserNotConfirmedException': {
      errorMessage = 'User has not yet been confirmed. Please check your email or text messages.';
      break;
    }
  }

  setState({errorMessage});
};

export const onSubmit = (props) => (values): void => {
  const {onLogin = () => {}} = props;
  const {password, username} = values;
  onLogin(username, password);
};

export const validate = (): any => {

};

export const renderLogo = (logo: JSX.Element, classes): JSX.Element => {
  if(logo) {
    return <div className={classes.logo}>{logo}</div>;
  }

  return null;
};

export const renderMessage = (message: string): JSX.Element => {
  if(message) {
    const classes = useStyles();
    return <div className={classes.errorMessage}>{message}</div>;
  }

  return null;
};

export const LoginView = (props: LoginViewProps) => {
  const {
    logo,
    name = 'login',
    onSignup = () => {},
    title = 'Login'
  } = props;

  // Initial state
  const [state, setState] = useState({
    account: {
      password: '',
      username: ''
    },
    errorMessage: null
  });
  const {account, errorMessage} = state;

  const classes = useStyles();

  useFlux(LoginConstants.SUCCESS, onSuccess);
  useFlux(LoginConstants.FAILED, onFailure(setState));

  return (
    <PageView name={name} title={title}>
      <div className="row">
        <div className="col">
          <PageHeader title="Welcome Back">
            Welcome back sign-in now there is lot of new stuff waiting for you.
          </PageHeader>
        </div>
      </div>

      <div className="row justify-content-center">
        <div className="col-md-8 col-lg-6">
          <Card className={classes.card}>
            <CardContent>
              {renderLogo(logo, classes)}
              {renderMessage(errorMessage)}
              <Form
                onSubmit={onSubmit(props)}
                initialValues={account}
                validate={validate}
                validateOnBlur>
                <div className="container">
                  <div className="row">
                    <div className="col">
                      <TextField label="Username" name="username" type="text" />
                    </div>
                  </div>
                  <div className="row">
                    <div className="col">
                      <TextField label="Password" name="password" type="password" />
                    </div>
                  </div>
                  <div className="row">
                    <div className="col">
                      <p className={classes.forgot}>Forgot password?</p>
                    </div>
                  </div>
                  <div className="row justify-content-end">
                    <div className="col-xs-auto">
                      <Button className={classes.btn} color="default" size="large" onClick={onSignup} variant="text">
                        Create new account
                      </Button>
                      <Button className={classes.btn} color="primary" size="large" type="submit" variant="contained">
                        Sign in
                      </Button>
                    </div>
                  </div>
                </div>
              </Form>
            </CardContent>
          </Card>
        </div>
      </div>
    </PageView >
  );
};

export default LoginView;
