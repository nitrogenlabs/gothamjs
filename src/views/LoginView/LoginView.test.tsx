import * as React from 'react';
import * as renderer from 'react-test-renderer';

import {LoginView} from './LoginView';

describe('LoginView', () => {
  let rendered;

  beforeAll(() => {
    // Render
    rendered = renderer.create(<LoginView />);
  });

  it('should render', () => expect(rendered).toBeDefined());

  it('should match snapshot', () => expect(rendered).toMatchSnapshot());
});
