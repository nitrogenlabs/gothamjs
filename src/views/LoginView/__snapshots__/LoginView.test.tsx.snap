// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginView should match snapshot 1`] = `
<div
  className="PageViewBase-view-7"
>
  <div
    className="PageViewBase-toolbar-6"
  />
  <div
    className="MuiGrid-container-8 MuiGrid-direction-xs-column-11"
  >
    <div
      className="PageViewBase-title-5"
    >
      <h1
        className="MuiTypography-root-104 MuiTypography-display1-108"
      >
        Login
      </h1>
    </div>
    <div
      className="MuiGrid-item-9 MuiGrid-grid-sm-12-61"
    >
      <div
        className="text-center"
      >
        <h1>
          Welcome Back
        </h1>
        <p
          className="subtitle"
        >
          Welcome back sign-in now there is lot of new stuff waiting for you.
        </p>
      </div>
    </div>
    <div
      className="MuiGrid-item-9 MuiGrid-grid-xs-12-47"
    >
      <div
        className="MuiGrid-container-8 MuiGrid-align-items-xs-center-16 MuiGrid-justify-xs-center-25"
      >
        <div
          className="MuiPaper-root-129 MuiPaper-elevation1-132 MuiPaper-rounded-130 MuiCard-root-128 LoginViewBase-card-2"
        >
          <div
            className="MuiCardContent-root-156"
          >
            <form
              className="FormBase-form-157"
              onSubmit={[Function]}
            >
              <div
                className="MuiGrid-item-9 MuiGrid-grid-xs-12-47"
              >
                <div
                  className="MuiFormControl-root-158 MuiFormControl-fullWidth-161"
                >
                  <label
                    className="MuiFormLabel-root-167 MuiInputLabel-root-162 MuiInputLabel-formControl-163 MuiInputLabel-animated-166"
                    data-shrink={false}
                  >
                    Username
                  </label>
                  <div
                    className="MuiInput-root-172 MuiInput-fullWidth-179 MuiInput-formControl-173 MuiInput-underline-176"
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInput-input-180"
                      disabled={false}
                      name="username"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-item-9 MuiGrid-grid-xs-12-47"
              >
                <div
                  className="MuiFormControl-root-158 MuiFormControl-fullWidth-161"
                >
                  <label
                    className="MuiFormLabel-root-167 MuiInputLabel-root-162 MuiInputLabel-formControl-163 MuiInputLabel-animated-166"
                    data-shrink={false}
                  >
                    Password
                  </label>
                  <div
                    className="MuiInput-root-172 MuiInput-fullWidth-179 MuiInput-formControl-173 MuiInput-underline-176"
                  >
                    <input
                      aria-invalid={false}
                      className="MuiInput-input-180 MuiInput-inputType-183"
                      disabled={false}
                      name="password"
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      type="password"
                      value=""
                    />
                  </div>
                </div>
              </div>
              <div
                className="MuiGrid-container-8 MuiGrid-justify-xs-flex-end-26 LoginViewBase-forgot-3"
              >
                <p
                  className="forgetPass"
                >
                  Forgot password???
                </p>
              </div>
              <div
                className="MuiGrid-container-8 MuiGrid-align-items-xs-flex-end-18 MuiGrid-justify-xs-flex-end-26 LoginViewBase-buttons-1"
              >
                <button
                  className="MuiButtonBase-root-209 MuiButton-root-185 MuiButton-text-187 MuiButton-flat-190 MuiButton-sizeLarge-207"
                  disabled={false}
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="submit"
                  value=""
                >
                  <span
                    className="MuiButton-label-186"
                  >
                    New Account
                  </span>
                  <span
                    className="MuiTouchRipple-root-212"
                  />
                </button>
                <button
                  className="MuiButtonBase-root-209 MuiButton-root-185 MuiButton-contained-194 MuiButton-containedPrimary-195 MuiButton-raised-197 MuiButton-raisedPrimary-198 MuiButton-sizeLarge-207"
                  disabled={false}
                  onBlur={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onKeyUp={[Function]}
                  onMouseDown={[Function]}
                  onMouseLeave={[Function]}
                  onMouseUp={[Function]}
                  onTouchEnd={[Function]}
                  onTouchMove={[Function]}
                  onTouchStart={[Function]}
                  tabIndex="0"
                  type="submit"
                  value=""
                >
                  <span
                    className="MuiButton-label-186"
                  >
                    Sign in
                  </span>
                  <span
                    className="MuiTouchRipple-root-212"
                  />
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
